src/pybamm/simulation.py

```diff
@@ -877,19 +878,28 @@ def solve(
                 # At the final step of the inner loop we save the cycle
                 if len(steps) > 0:
                     # Check for EmptySolution
-                    if all(isinstance(step, pybamm.EmptySolution) for step in steps):
+                    if all(
+                        isinstance(step_solution, pybamm.EmptySolution)
+                        for step_solution in steps
+                    ):
                         if len(steps) == 1:
-                            raise pybamm.SolverError(
-                                f"Step '{step_str}' is infeasible "
-                                "due to exceeded bounds at initial conditions. "
-                                "If this step is part of a longer cycle, "
-                                "round brackets should be used to indicate this, "
-                                "e.g.:\n pybamm.Experiment([(\n"
-                                "\tDischarge at C/5 for 10 hours or until 3.3 V,\n"
-                                "\tCharge at 1 A until 4.1 V,\n"
-                                "\tHold at 4.1 V until 10 mA\n"
-                                "])"
-                            )
+                            if step.skip_ok:
+                                pybamm.logger.warning(
+                                    f"Step '{step_str}' is infeasible at initial conditions, but skip_ok is True. Skipping step."
+                                )
+                                continue
+                            else:
+                                raise pybamm.SolverError(
+                                    f"Step '{step_str}' is infeasible "
+                                    "due to exceeded bounds at initial conditions. "
+                                    "If this step is part of a longer cycle, "
+                                    "round brackets should be used to indicate this, "
+                                    "e.g.:\n pybamm.Experiment([(\n"
+                                    "\tDischarge at C/5 for 10 hours or until 3.3 V,\n"
+                                    "\tCharge at 1 A until 4.1 V,\n"
+                                    "\tHold at 4.1 V until 10 mA\n"
+                                    "])"
+                                )
                         else:
                             this_cycle = self.experiment.cycles[cycle_num - 1]
                             raise pybamm.SolverError(
```

Comment:
If all experiments in a cycle have `skip_ok = True`, should we still raise this error?

Reply:
I kinda think we should raise an error in that case, but maybe we should raise a slightly different error. What do you think?

Reply:
Yeah, it's not obvious to me how we should handle this. If every step in a cycle is skipped it suggests something is probably wrong, but on the other hand, it is what they're asking for. I think raising a slightly different error message is reasonable

Reply:
I think we can figure that out in conjunction with #4840. This is intended as more of a patch